# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/openrgb/hotfix-corsairpro-ddr5.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- OpenRGB-release_0.8/Controllers/CorsairVengeanceProController/CorsairVengeanceProControllerDetect.cpp.vanilla	2023-05-11 17:19:44.275623940 +0200
+++ OpenRGB-release_0.8/Controllers/CorsairVengeanceProController/CorsairVengeanceProControllerDetect.cpp	2023-05-11 17:39:40.981710348 +0200
@@ -33,7 +33,7 @@
 
         res = bus->i2c_smbus_read_byte_data(address, 0x43);
 
-        if (res != 0x1C)
+        if (res != 0x1B && res != 0x1C)
         {
             pass = false;
             LOG_DEBUG("[%s] Failed: was expecting 0x1C got %02X", CORSAIR_VENGEANCE_RGB_PRO_NAME, res);
@@ -81,6 +81,16 @@
             {
                 if(TestForCorsairVengeanceProController(busses[bus], addr))
                 {
+                    CorsairVengeanceProController*     new_controller    = new CorsairVengeanceProController(busses[bus], addr);
+                    RGBController_CorsairVengeancePro* new_rgbcontroller = new RGBController_CorsairVengeancePro(new_controller);
+
+                    ResourceManager::get()->RegisterRGBController(new_rgbcontroller);
+                }
+            }
+            for(unsigned char addr = 0x18; addr <= 0x1F; addr++)
+            {
+                if(TestForCorsairVengeanceProController(busses[bus], addr))
+                {
                     CorsairVengeanceProController*     new_controller    = new CorsairVengeanceProController(busses[bus], addr);
                     RGBController_CorsairVengeancePro* new_rgbcontroller = new RGBController_CorsairVengeancePro(new_controller);
 
